rustls-libssl/tests/runner.rs

```diff
@@ -327,6 +327,119 @@ fn server() {
     assert_eq!(openssl_output, rustls_output);
 }
 
+const NGINX_LOG_LEVEL: &str = "info";
+
+#[test]
+#[ignore]
+fn nginx() {
+    fs::create_dir_all("target/nginx-tmp/basic/html").unwrap();
+    fs::write(
+        "target/nginx-tmp/basic/server.conf",
+        "
+daemon off;
+master_process off;
+pid nginx.pid;
+
+events {
+}
+
+http {
+    ssl_protocols TLSv1.2 TLSv1.3;
+    access_log access.log;
+
+    server {
+        listen 8443 ssl;
+        server_name localhost;
+        ssl_certificate ../../../test-ca/rsa/server.cert;
+        ssl_certificate_key ../../../test-ca/rsa/server.key;
+
+        location = / {
+            return 200 \"hello world\n\";
+        }
+
+        location /ssl-agreed {
+            return 200 \"protocol:$ssl_protocol,cipher:$ssl_cipher\n\";
+        }
+
+        location /ssl-server-name {
+            return 200 \"server-name:$ssl_server_name\n\";
+        }
+
+        location /ssl-was-reused {
+            return 200 \"reused:$ssl_session_reused\n\";
+        }
+
+        # not currently implemented:
+        location /ssl-offer {
+            return 200 \"ciphers:$ssl_ciphers,curves:$ssl_curves\n\";
+        }
+
+        location /ssl-early-data {
+            return 200 \"early-data:$ssl_early_data\n\";
+        }
+
+        location /ssl-client-auth {
+            return 200 \"s-dn:$ssl_client_s_dn\ni-dn:$ssl_client_i_dn\nserial:$ssl_client_serial\nfp:$ssl_client_fingerprint\nverify:$ssl_client_verify\nv-start:$ssl_client_v_start\nv-end:$ssl_client_v_end\nv-remain:$ssl_client_v_remain\ncert:\n$ssl_client_cert\n\";
+        }
+    }
+}
```

Comment:
WDYT about making this a separate asset file and using `include_string!`? It would allow editors that understand nginx config to process this nicer than as an inline string context.

Reply:
Yes good idea.